sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessor.cs

```diff
@@ -30,6 +28,21 @@ public class EventProcessor<T> where T : BasePartitionProcessor, new()
         /// <summary>The primitive for synchronizing access during start and close operations.</summary>
         private readonly SemaphoreSlim _runningTaskSemaphore = new SemaphoreSlim(1, 1);
 
+        /// <summary>The primitive for synchronizing access during start and set handler operations.</summary>
+        private readonly object _setHandlerMutex = new object();
```

Comment:
I don't think it would be worth having a different mutex for each one of the four properties. The operations it locks are simple and shouldn't block for too long.